### DISCUSSION TITLE
v19.5.0-rc1

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3025] [Releases]
## Highlights
This is a big release, that's why we decided to move away from the 19.4.x versioning and go to 19.5.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 19.5.0. Tests are more than welcome now but be extra careful with production environment.

What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.

Also the M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

## Changelog
* Update title size of unsubscription email by @luigifab in https://github.com/OpenMage/magento-lts/pull/2722
* Require a parent category to add a new sub category by @luigifab in https://github.com/OpenMage/magento-lts/pull/2716
* Version bump for next release by @fballiano in https://github.com/OpenMage/magento-lts/pull/2769
* Use store data for products of order items by @luigifab in https://github.com/OpenMage/magento-lts/pull/2723
* Fix error when payment methods have been deleted by @sreichel in https://github.com/OpenMage/magento-lts/pull/2772
* Fixed sort in Manage Tax Rates grid by @sreichel in https://github.com/OpenMage/magento-lts/pull/2757
* Use default paths for config files by @sreichel in https://github.com/OpenMage/magento-lts/pull/2765
* Moved phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier to composer by @fballiano in https://github.com/OpenMage/magento-lts/pull/2411
* Fixes workflow issues, ref #2770 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2773
* Added Cm_Redis files to .gitignore by @sreichel in https://github.com/OpenMage/magento-lts/pull/2779
* Hotfix: broken workflow by @sreichel in https://github.com/OpenMage/magento-lts/pull/2778
* Removed unreachable code by @sreichel in https://github.com/OpenMage/magento-lts/pull/2775
* Avoid to use unavailable $data var in Curl HTTP Client by @maximehuran in https://github.com/OpenMage/magento-lts/pull/2785
* phpstan: added lib/Mage and lib/Magento by @sreichel in https://github.com/OpenMage/magento-lts/pull/2780
* Updated DOCblocks (fixed param null) by @sreichel in https://github.com/OpenMage/magento-lts/pull/2776
* Fixed baseline, ref #2785 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2789
* PHPStan: removed excluded directories by @sreichel in https://github.com/OpenMage/magento-lts/pull/2790
* Reverted autoloader patch by @sreichel in https://github.com/OpenMage/magento-lts/pull/2791
* PHPStan: Level 0 update by @sreichel in https://github.com/OpenMage/magento-lts/pull/2794
* Check $sessionData is an array in Mage_Captcha_Model_Zend by @fballiano in https://github.com/OpenMage/magento-lts/pull/2804
* Moved null-byte fix from lib/Zend to lib/Magento by @sreichel in https://github.com/OpenMage/magento-lts/pull/2807
* Updated phpstan 1.9.3 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2808
* PHPStan: updated lib/Varien by @sreichel in https://github.com/OpenMage/magento-lts/pull/2795
* Replaced MySql4 classes in installer by @sreichel in https://github.com/OpenMage/magento-lts/pull/2797
* Updated phpdocs by @sreichel in https://github.com/OpenMage/magento-lts/pull/2796
* Sync v19 v20 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2810
* Created a release builder workflow by @fballiano in https://github.com/OpenMage/magento-lts/pull/2165
* phpstan: Mage.php by @sreichel in https://github.com/OpenMage/magento-lts/pull/2819
* phpstan: Mage_Poll by @sreichel in https://github.com/OpenMage/magento-lts/pull/2816
* phpstan: Mage_Rss by @sreichel in https://github.com/OpenMage/magento-lts/pull/2817
* phpstan: Mage_Page by @sreichel in https://github.com/OpenMage/magento-lts/pull/2820
* Add confirm dialog to critical massactions by @sreichel in https://github.com/OpenMage/magento-lts/pull/2814
* Added cweagans/composer-patches - prepare for ZF1Future :rocket:  by @sreichel in https://github.com/OpenMage/magento-lts/pull/2822
* [Backport] Remove documentation hints, ref #1536 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2815
* phpstan: Mage_Cms by @sreichel in https://github.com/OpenMage/magento-lts/pull/2818
* Optimisation for Varien_Object::_addFullNames by @AGelzer in https://github.com/OpenMage/magento-lts/pull/2821
* Fix passing null and array to string conversion error by @sreichel in https://github.com/OpenMage/magento-lts/pull/2824
* [php8.1] deprecated PDOStatement::fetch, ref #1812 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2805
* phpstan: Sitemap, Newsletter, ... by @sreichel in https://github.com/OpenMage/magento-lts/pull/2823
* phpstan: added missing returns by @sreichel in https://github.com/OpenMage/magento-lts/pull/2832
* Replace lib/Zend with shardj/zf1-future :rocket:  by @sreichel in https://github.com/OpenMage/magento-lts/pull/2827
* phpstan: fixes "Call to function is_null ..." by @sreichel in https://github.com/OpenMage/magento-lts/pull/2831
* Sonar: fixed path to lib/Zend by @sreichel in https://github.com/OpenMage/magento-lts/pull/2834
* Fixed bugs for admin save base urls by @sreichel in https://github.com/OpenMage/magento-lts/pull/2800
* Added getApplyTo() to Mage_Eav_Model_Entity_Attribute_Abstract. ref #2829 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2836
* Removed Mage_PageCache by @sreichel in https://github.com/OpenMage/magento-lts/pull/2813
* phpstan: step back to level 4 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2837
* Version bump by @fballiano in https://github.com/OpenMage/magento-lts/pull/2835
* phpstan: Change OpenMage version compare by @sreichel in https://github.com/OpenMage/magento-lts/pull/2839
* phpstan: working on level 3 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2840
* Added dependabot config by @sreichel in https://github.com/OpenMage/magento-lts/pull/2841
* Moved note about PHP7.2 since it is not supported anymore by @fballiano in https://github.com/OpenMage/magento-lts/pull/2842
* Bump tj-actions/changed-files from 34 to 35 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2843
* Bump symfonycorp/security-checker-action from 4 to 5 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2845
* Bump EnricoMi/publish-unit-test-result-action from 1.6 to 1.40 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2846
* Bump pelago/emogrifier from 6.0.0 to 7.0.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2844
* Added helper for admin button onclick actions by @sreichel in https://github.com/OpenMage/magento-lts/pull/2784
* Added shell/ to checks by @sreichel in https://github.com/OpenMage/magento-lts/pull/2848
* autoload without hiding errors by @Flyingmana in https://github.com/OpenMage/magento-lts/pull/2300
* Use correct code for Greece VAT validation by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2849
* Updated lib/Varien for PHP8.1 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2802
* Added .dist and .neon to "deny from all" in .htaccess by @fballiano in https://github.com/OpenMage/magento-lts/pull/2852
* Added notes about composer library/modules to README (for 19.5.x and 20.1.x) by @fballiano in https://github.com/OpenMage/magento-lts/pull/2851
* phpstan: remove one diff between v19/20 baseline by @sreichel in https://github.com/OpenMage/magento-lts/pull/2855
* Hotfix: php7 has no return type "mixed" by @sreichel in https://github.com/OpenMage/magento-lts/pull/2856
* Add translation helper shell script by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2332
* PHPMD: added basic config by @sreichel in https://github.com/OpenMage/magento-lts/pull/2771
* Load dev shell scripts as composer module by @sreichel in https://github.com/OpenMage/magento-lts/pull/2853
* Fixed tag aggregation indexer query by @fballiano in https://github.com/OpenMage/magento-lts/pull/2858
* Updated workflow: run when files are deleted by @sreichel in https://github.com/OpenMage/magento-lts/pull/2860
* Rewrote Mage_Reports_Model_Resource_Review_Product_Collection/Mage_Reports_Model_Resource_Order_Collection queries for a correct use of Zend_Db_Expr by @fballiano in https://github.com/OpenMage/magento-lts/pull/2864
* Backport 2271, removed lib/flex by @fballiano in https://github.com/OpenMage/magento-lts/pull/2862
* Updated copyright blocks by @sreichel in https://github.com/OpenMage/magento-lts/pull/2866
* Updated autoloader, ref #2300 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2867
* Adding useful feedback to Gd2.php exceptions by @loekvangool in https://github.com/OpenMage/magento-lts/pull/1339
* Added ddev command shortcuts by @sreichel in https://github.com/OpenMage/magento-lts/pull/2868
* Use github URL for patch files by @sreichel in https://github.com/OpenMage/magento-lts/pull/2871
* Remove "was" from error messages by @loekvangool in https://github.com/OpenMage/magento-lts/pull/2869
* Add autocomplete attribute to known password fields. by @rfeese in https://github.com/OpenMage/magento-lts/pull/2700
* Create codeql-analysis.yml by @Flyingmana in https://github.com/OpenMage/magento-lts/pull/2644
* Cast types, ref #735 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2872
* Fix error on add new contributor by @AGelzer in https://github.com/OpenMage/magento-lts/pull/2877
* Fix for #2874 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2878
* Bump colinmollenhour/cache-backend-redis from 1.14.4 to 1.15.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2881
* Exception on add reserved catalog product attribute by @AGelzer in https://github.com/OpenMage/magento-lts/pull/2875
* rector: automatic code refactoring (part 1) by @sreichel in https://github.com/OpenMage/magento-lts/pull/2879
* Added DEV_PHP_STRICT env variable by @sreichel in https://github.com/OpenMage/magento-lts/pull/2882
* Updated PHPStan to 1.9.6 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2887
* Fixes passing null to strlen() inside escapeHtml() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2884
* Ensure null values are not passed to trim() by @mattdavenport in https://github.com/OpenMage/magento-lts/pull/2889
* Removed Mage_Backup by @sreichel in https://github.com/OpenMage/magento-lts/pull/2811
* Require Mage_Backup and Mage_PageCache on v19's release builder by @fballiano in https://github.com/OpenMage/magento-lts/pull/2890
* Added getter/setter for getCouponCode() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2825
* Added getter for Customer::getPassword(), ref #2586 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2891
* Removed info about Magento versions merging by @fballiano in https://github.com/OpenMage/magento-lts/pull/2894
* Use method instead of _getData(), ref #2825 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2892
* Updated README.md by @sreichel in https://github.com/OpenMage/magento-lts/pull/2897
* phpstan/plugin version update by @sreichel in https://github.com/OpenMage/magento-lts/pull/2896
* Better exceptions for file upload by @fballiano in https://github.com/OpenMage/magento-lts/pull/2902
* Added default return string to getParam() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2893
* Ensure count() is not called on a string by @mattdavenport in https://github.com/OpenMage/magento-lts/pull/2888
* Cast return of getClass() to string by @sreichel in https://github.com/OpenMage/magento-lts/pull/2905
* Updated calls to strlen(), ref #2586 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2898
* Bump symplify/vendor-patches from 11.1.24 to 11.1.31.72 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2911
* Bump phpstan/phpstan from 1.9.7 to 1.9.8 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2913
* Bump dealerdirect/phpcodesniffer-composer-installer from 0.7.2 to 1.0.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2912
* Ensure null value not passed to str_replace by @mattdavenport in https://github.com/OpenMage/magento-lts/pull/2901
* Fix Use Base Class Instead of Child Class when Referencing Constants by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2904
* Removed unused autoloaders Mage_Autoload_Simple and Magento\Autoload\Simple by @fballiano in https://github.com/OpenMage/magento-lts/pull/2859
* [PHP 8.1] Fix passing null to non-nullable internal function params by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2586
* Fix searching for utf8 characters by @sreichel in https://github.com/OpenMage/magento-lts/pull/2919
* remove scriptaculous/effects.js dependency in js/prototype/validation.js by @empiricompany in https://github.com/OpenMage/magento-lts/pull/2609
* Better error handling when XML file cannot be parsed, ref #1372 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2923
* RWD theme: removed border bottom from h1/h2 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2920
* getAttributeRawValue return false if value is '0' by @bastienlm in https://github.com/OpenMage/magento-lts/pull/572
* Added info about #572 to README by @fballiano in https://github.com/OpenMage/magento-lts/pull/2926
* New event "order_status_changed_before_save", ref #1128 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2927
* Generalized currency formatting method formatPriceTxt() for use in or… by @kiatng in https://github.com/OpenMage/magento-lts/pull/2924
* Add some specific debugging hint to email queue recipient empty by @seansan in https://github.com/OpenMage/magento-lts/pull/1386
* Remove unmatched error from phpstan baseline by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2929
* Limit decrypt to 256 characters by @luigifab in https://github.com/OpenMage/magento-lts/pull/2685
* Documenting all shell/indexer.php arguments with dashes by @loekvangool in https://github.com/OpenMage/magento-lts/pull/2930
* Fix error when calling shell script from browser by @Caprico85 in https://github.com/OpenMage/magento-lts/pull/2674
* Fix unlink no such file or directory by @luigifab in https://github.com/OpenMage/magento-lts/pull/2675
* Check ACL for customer link of orders by @luigifab in https://github.com/OpenMage/magento-lts/pull/2933
* Add Events around Zend_Email by @rjocoleman in https://github.com/OpenMage/magento-lts/pull/1438
* PHP8.1 Apply latest ZF1F changes to patch OM-918 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2936
* Allow rewrite Varien_Image by @luigifab in https://github.com/OpenMage/magento-lts/pull/2666
* PHP8.0 Fixes passing null as 1st parameter to mktime() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2938
* Fixes typo in observer by @sreichel in https://github.com/OpenMage/magento-lts/pull/2942
* PHP8.1 Fixes passing null to trim() - for non-existing payment methods by @sreichel in https://github.com/OpenMage/magento-lts/pull/2939
* Bump phpunit/phpunit from 9.5.27 to 9.5.28 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2947
* Bump symplify/vendor-patches from 11.1.31.72 to 11.2.0.72 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2948
* Bump phpstan/phpstan from 1.9.8 to 1.9.11 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2949
* Replace strftime with date by @luigifab in https://github.com/OpenMage/magento-lts/pull/2934
* Removed app/code/core/Zend by @fballiano in https://github.com/OpenMage/magento-lts/pull/2861
* Replace deprecated gmstrftime with gmdate by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2951
* Adds polyfills for PHP 7.4-8.1 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2946
* Fix less than typo by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2952
* Disable cash on delivery for virtual orders by @luigifab in https://github.com/OpenMage/magento-lts/pull/2953
* Reduce getId calls by @luigifab in https://github.com/OpenMage/magento-lts/pull/2957
* Ascending sort for themes/packages dropdown by @sreichel in https://github.com/OpenMage/magento-lts/pull/2967
* Fix for mcrypt_compat 2.0.4 by @luigifab in https://github.com/OpenMage/magento-lts/pull/2968
* Updated PHPStan to 1.9.13 and ZF1Future to 1.22 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2960
* Brexit of Great-Britain by @luigifab in https://github.com/OpenMage/magento-lts/pull/2971
* Improving query performance for coupon code check by @lemundo-team in https://github.com/OpenMage/magento-lts/pull/1110
* Set dependabot updates from weekly to daily by @sreichel in https://github.com/OpenMage/magento-lts/pull/2970
* Keep current selection for Related, Up-sells, Cross-sells, Associated grids by @luigifab in https://github.com/OpenMage/magento-lts/pull/2715
* Bump colinmollenhour/cache-backend-redis from 1.15.0 to 1.16.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2972
* Added counter for input fields with maximum-length attribute by @sreichel in https://github.com/OpenMage/magento-lts/pull/2950
* Add ifconfig for layout updates by @luigifab in https://github.com/OpenMage/magento-lts/pull/2974
* Add template only if necessary (ups with Mage_Usa) by @luigifab in https://github.com/OpenMage/magento-lts/pull/2975
* Bump phpstan/phpstan from 1.9.13 to 1.9.14 by @dependabot in https://github.com/OpenMage/magento-lts/pull/2983
* Update jscolor to 1.4.5 by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2976
* Remove version check from api.php, get.php, index.php, install.php by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2986
* Wrap WHERE condition in parens by @mattdavenport in https://github.com/OpenMage/magento-lts/pull/2965
* Install composer dependencies in gitpod by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2989
* Moved patch files to repo by @sreichel in https://github.com/OpenMage/magento-lts/pull/2985
* Added a forced refresh if the user modifies the minicart while being on the cart page by @fballiano in https://github.com/OpenMage/magento-lts/pull/2990
* Small Typo Fix otions => options by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2994
* Update README install instructions by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2981
* Clear hasDataChanges flag after loading collection items by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3000
* Fix HEADERS ALREADY SENT from Mage_Core_Block_Template by @luigifab in https://github.com/OpenMage/magento-lts/pull/2978
* Bugfixes addto.phtml by @loekvangool in https://github.com/OpenMage/magento-lts/pull/3002
* Fixed bug in prototypejs by @fballiano in https://github.com/OpenMage/magento-lts/pull/3003
* Resolve getPasswordTimestamp join issues by @alexkirsch in https://github.com/OpenMage/magento-lts/pull/3010
* Head.php to HTML5 by @loekvangool in https://github.com/OpenMage/magento-lts/pull/3006
* Fix phpseclib class namespace in Varien_Io_Sftp by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3008
* Bump phpunit/phpunit from 9.5.28 to 9.6.3 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3013
* Fix bug in admin interface where Bundle Option Is Required setting is not loaded. by @rfeese in https://github.com/OpenMage/magento-lts/pull/3014
* Backported #1600: removed M1 legacy themes by @fballiano in https://github.com/OpenMage/magento-lts/pull/2863
* Bump phpstan/phpstan from 1.9.14 to 1.9.16 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3016
* Bump symplify/vendor-patches from 11.2.0.72 to 11.2.3 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3015
* Reduce getId calls by @luigifab in https://github.com/OpenMage/magento-lts/pull/3007
* Bump phpstan/phpstan from 1.9.16 to 1.9.17 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3018
* Fix bug in admin interface where Bundle Option Select User Defined Qty setting is not loaded. by @rfeese in https://github.com/OpenMage/magento-lts/pull/3020
* Remove minOccurs attr from WSDL message part by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3022
* Fix position sort with flat tables when no category filter is applied by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2637

## New Contributors
* @AGelzer made their first contribution in https://github.com/OpenMage/magento-lts/pull/2821
* @dependabot made their first contribution in https://github.com/OpenMage/magento-lts/pull/2843
* @rfeese made their first contribution in https://github.com/OpenMage/magento-lts/pull/2700
* @alexkirsch made their first contribution in https://github.com/OpenMage/magento-lts/pull/3010

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v19.4.20...v19.5.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v19.5.0-rc1'>v19.5.0-rc1</a>.</em>

**Comment by luigifab** [comment]
I tested to `composer update` from 20.0.20 to 20.1.0-rc1.
I think but I'm not sure, but `composer` haven't removed `lib/Credis lib/Pelago lib/Zend lib/phpseclib lib/mcryptcompat`.

I also don't understand this new file `app/code/community/Cm/Cache/Backend/Redis.php`...
Perhaps I'm wrong, but I have removed it, and all seems to work.

After I checked (only) `v20.1.0-rc1.zip` and the following directories are empty:
 `vendor/symfony/string/Resources/bin`
 `vendor/colinmollenhour/cache-backend-redis/lib/Credis`
 `lib/Credis`

Not sure why.

**Reply by fballiano** [reply]
answering just to a small part, https://github.com/OpenMage/magento-lts#since-openmage-1950--2010 you can find a command to remove ib/Credis lib/Pelago lib/Zend lib/phpseclib lib/mcryptcompat, composer won't remove those.

**Reply by luigifab** [reply]
oh yeah! I didn't read the readme, only the changelog here!

**Reply by fballiano** [reply]
ye I should have mentioned it but it's not easy to keep track of everything

**Comment by luigifab** [comment]
Today I tested `v20.1.0-rc1` (this probably same for `19.5.0-rc1`) with PHP 8.2.
Here are the deprecated notices I have in [Sentry](https://github.com/luigifab/openmage-sentry):

![deprecated](https://user-images.githubusercontent.com/31816829/222791581-fe1be5d8-f508-425d-ae75-3bda8110f973.png)

We are almost there!

**Reply by elidrissidev** [reply]
Can you post all these deprecations with the filename and line number? I'd like to tackle them and that'd help a lot.

**Reply by luigifab** [reply]
Table.php, _app/code/core/Mage/Eav/Model/Entity/Attribute/Source/Table.php_, line 84, `if (strpos($value, ',')) {`

An idea:
```diff
@@ -80,6 +80,10 @@ class Mage_Eav_Model_Entity_Attribute_Source_Table extends Mage_Eav_Model_Entity
      */
     public function getOptionText($value)
     {
+        if ($value === null) {
+            return false;
+        }
+
         $isMultiple = false;
         if (strpos($value, ',')) {
             $isMultiple = true;
```

--

Source.php, _app/code/core/Mage/Catalog/Model/Resource/Product/Indexer/Eav/Source.php_, line 238, `$values = array_unique(explode(',', $row['value']));`

--

Abstract.php, _app/code/core/Mage/Core/Model/Resource/Helper/Abstract.php_, line 309, `'identity' => stripos($column['extra'], 'auto_increment') !== false`

--

Abstract.php, _app/code/core/Mage/Core/Model/Resource/Db/Abstract.php_, line 666, `$select->where($field . '=?', trim($data->getData($field)));`

--

Flat.php, _app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php_, line 670, `$column['DEFAULT'] = trim($column['DEFAULT'], "' ");`

**Reply by AGelzer** [reply]
`if (strpos($value, ',')) {`

`$value` value must be of type string

```diff
- if (strpos($value, ',')) {
+  if (is_string($value) && strpos($value, ',')) {
```

or

```diff
- if (strpos($value, ',')) {
+  if (strpos((string) $value, ',')) {
```

**Comment by luigifab** [comment]
I found the following incompatible change in `v20.1.0-rc1` (probably also apply to `v19.5.0-rc1`).
Before updating, I already use [zf1-future](https://github.com/Shardj/zf1-future) 1.22.0 (commit 5446af3).

Before we can:
`->columns('action_key, count(*) as count')`
Now we can't. This must be:
`->columns(['action_key', 'count(*) as count'])`

Sentry report:
![image](https://user-images.githubusercontent.com/31816829/222792838-d1bfcc15-31fc-4d2e-b1d9-c86078691dd5.png)

---

Before we can:
`$columns['average_days'] = 'CASE WHEN ... END';
$orders->getSelect()->columns($columns);`
Now we can't. This must be:
`$columns['average_days'] = new Zend_Db_Expr('CASE WHEN ... END');`


Sentry report:
![image](https://user-images.githubusercontent.com/31816829/222793213-1cc5cfb2-5586-47ce-a770-236990e453af.png)

**Reply by addison74** [reply]
If it concerns ZF1-Future, it must be reported in its repository. PR can also be created there, I found that they are approved quite quickly.

**Reply by luigifab** [reply]
I suppose this come from OpenMage, but I haven't searched.

**Reply by elidrissidev** [reply]
For the second issue, it's a security change that was added to zf1-future, it was already discussed in the PR

**Reply by luigifab** [reply]
Which PR? I didn't see it, or I forgot it.

**Reply by elidrissidev** [reply]
I forgot you were not part of the reviewers. It was discussed in https://github.com/OpenMage/magento-lts/pull/2861.

**Reply by luigifab** [reply]
Okay, so there is no problem.
Perhaps add this information to the final releases.

**Comment by luigifab** [comment]
There are also two new bugs:

With #2902, each time I save a category without any changes, I have: `No file was uploaded`
![image](https://user-images.githubusercontent.com/31816829/222805874-fa7e3715-e0dc-4340-a9f2-e06dac36cde8.png)

I fixed with (but not sure if it's good or not):
```diff
--- a/web/lib/Varien/File/Uploader.php
+++ b/web/lib/Varien/File/Uploader.php
@@ -161,10 +161,10 @@ class Varien_File_Uploader
         $this->_setUploadFileId($fileId);
         if (empty($this->_file['tmp_name']) || !file_exists($this->_file['tmp_name'])) {
             $errorCode = $this->_file['error'] ?? 0;
+            $code = empty($this->_file['tmp_name']) ? self::TMP_NAME_EMPTY : 0;
             if ($errorCode && isset(self::UPLOAD_ERRORS[$errorCode])) {
-                throw new Exception(self::UPLOAD_ERRORS[$errorCode]);
+                throw new Exception(self::UPLOAD_ERRORS[$errorCode], $code);
             }
-            $code = empty($this->_file['tmp_name']) ? self::TMP_NAME_EMPTY : 0;
             throw new Exception('File was not uploaded.', $code);
         } else {
             $this->_fileExists = true;
```
---

With #2939, I have tons of: `Unkown payment method with code ...`
![image](https://user-images.githubusercontent.com/31816829/222806248-89926f32-d33e-4c97-a52b-e9c3cd792106.png)

I removed the `Mage::log`.

**Reply by fballiano** [reply]
for #2939 I think you could change the payment method on the DB for old orders?

**Reply by luigifab** [reply]
Some of errors are generated by `Mage::helper('payment')->getRecurringProfileMethods()` or `Mage::helper('payment')->getPaymentMethodList(true)` for some grids.
I prefer to ignore the error log, and do nothing :)
Yes perhaps I can search in db and update it.

**Reply by fballiano** [reply]
I always do that when I remove a payment method from a customer's website

**Reply by fballiano** [reply]
maybe we could remove the log but I don't know... I think it's better to have the log and cleanup the DB, since a non-existant payment method can create other problems.

**Reply by fballiano** [reply]
regarding the first problem (saving category) I can't reproduce on a vanilla install, not on a client's website

**Reply by addison74** [reply]
It is not right at all that when you save a category without making any changes, to get that error message.  I will check if I can reproduce this issue. In #2902 I don't see any changes that generate this issue, only all the errors that can occur related to loading images are logged.

**Reply by luigifab** [reply]
:'( I'm sad, the stack trace is only in _app/code/core_ for the error when I save a category... I will check again later.
My change ($code) is for:

https://github.com/OpenMage/magento-lts/blob/ee46ff9b9a45809d5ebc7b09c13a91844f7c402c/app/code/core/Mage/Catalog/Model/Category/Attribute/Backend/Image.php#L60-L66

**Reply by luigifab** [reply]
The error is still here for me with commit [1e8fcd0](https://github.com/OpenMage/magento-lts/commit/1e8fcd0e155b4773d5531957f22cd1ac3e6f147c) - 19.5.0-rc2 (fresh git) in _exception.log_.

**Reply by elidrissidev** [reply]
@luigifab Yeah I'm able to see the exception running the rc2 release, I thought the exception shows up in the admin but it only logs it. I will investigate.

<details>
<summary>Full log</summary>

```log
2023-03-14T16:43:00+00:00 ERR (3): 
Exception: No file was uploaded in /usr/local/var/www/om/lib/Varien/File/Uploader.php:165
Stack trace:
#0 /usr/local/var/www/om/app/code/core/Mage/Catalog/Model/Category/Attribute/Backend/Image.php(54): Varien_File_Uploader->__construct('thumbnail')
#1 /usr/local/var/www/om/app/code/core/Mage/Eav/Model/Entity/Abstract.php(647): Mage_Catalog_Model_Category_Attribute_Backend_Image->afterSave(Object(Mage_Catalog_Model_Category))
#2 /usr/local/var/www/om/app/code/core/Mage/Eav/Model/Entity/Abstract.php(1647): Mage_Eav_Model_Entity_Abstract->walkAttributes('backend/afterSa...', Array)
#3 /usr/local/var/www/om/app/code/core/Mage/Catalog/Model/Resource/Category.php(225): Mage_Eav_Model_Entity_Abstract->_afterSave(Object(Mage_Catalog_Model_Category))
#4 /usr/local/var/www/om/app/code/core/Mage/Eav/Model/Entity/Abstract.php(1099): Mage_Catalog_Model_Resource_Category->_afterSave(Object(Mage_Catalog_Model_Category))
#5 /usr/local/var/www/om/app/code/core/Mage/Core/Model/Abstract.php(334): Mage_Eav_Model_Entity_Abstract->save(Object(Mage_Catalog_Model_Category))
#6 /usr/local/var/www/om/app/code/core/Mage/Adminhtml/controllers/Catalog/CategoryController.php(349): Mage_Core_Model_Abstract->save()
#7 /usr/local/var/www/om/app/code/core/Mage/Core/Controller/Varien/Action.php(428): Mage_Adminhtml_Catalog_CategoryController->saveAction()
#8 /usr/local/var/www/om/app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(262): Mage_Core_Controller_Varien_Action->dispatch('save')
#9 /usr/local/var/www/om/app/code/core/Mage/Core/Controller/Varien/Front.php(188): Mage_Core_Controller_Varien_Router_Standard->match(Object(Mage_Core_Controller_Request_Http))
#10 /usr/local/var/www/om/app/code/core/Mage/Core/Model/App.php(371): Mage_Core_Controller_Varien_Front->dispatch()
#11 /usr/local/var/www/om/app/Mage.php(748): Mage_Core_Model_App->run(Array)
#12 /usr/local/var/www/om/index.php(61): Mage::run('', 'store')
#13 {main}
```

</details>

**Reply by luigifab** [reply]
This is now fixed with #3086.
